### ISSUE TITLE
Colors are modified on PNG import

### ISSUE THREAD

**Issue by maeonian** [Scirra/Construct-3-bugs#5033]
## Problem description

When loading/importing PNG files to use within C3 (like tilesets or for characters, etc), there's a faint but noticable color modification/compression done to PNGs. It alters some of the colors on import. A work-around is to use GIFs instead, since JPGs get crunched up on import even worse than PNGs, but GIFs don't support smooth/faded alphas like PNGs do.

## Attach a .c3p

I've attached a c3p file renamed to a .zip, since I think C3 can load them that way. Hopefully you can see the differences I've mentioned. It's attached to the bottom, within the Details thing.

## Steps to reproduce

1. Create PNG
2. Import PNG
3. Add PNG / Sprite / object to layout and check the values with the original image

## Observed result

PNGs are slightly off in terms of colors

## Expected result

PNGs not to be compressed / manipulated on import. Other engines seem to render PNGs fine / don't color shift when dropped into their engine.

## More details

**Affected browsers/platforms:** 

Latest Firefox, Edge, and Chrome releases

**First affected release:** 

Noticed in r260 / latest stable as of writing

<details><summary>View details</summary>

Platform information

Browser: Firefox
Browser version: 91.0
Browser engine: Gecko
Context: browser
Operating system: Windows
Operating system version: 10
Device type: desktop
Device pixel ratio: 1.25
Logical CPU cores: 8
Approx. device memory: (unavailable)
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:91.0) Gecko/20100101 Firefox/91.0
C3 release: r260 (stable)
Language setting: en-US
Local storage

Storage quota (approx): 2 gb
Storage usage (approx): 18 mb (0.9%)
Persistant storage: No
Browser support notes

This list contains missing features that are not required, but could improve performance or user experience if supported.

    Rendering multiple on-screen Layout Views is slow in Firefox due to bug 1163426
    The <dialog> element is not supported. A polyfill is in use.
    The Clipboard API is not supported. Some clipboard features may be unavailable.
    ImageBitmapOptions is not supported. Texture loading performance may be degraded.
    Determining input device capabilities is not supported.

WebGL information

Version string: WebGL 2.0
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: no
Supports highp precision: yes
Vendor: Google Inc.
Renderer: ANGLE (Radeon R9 200 Series Direct3D11 vs_5_0 ps_5_0)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

    EXT_color_buffer_float
    EXT_float_blend
    EXT_texture_compression_bptc
    EXT_texture_compression_rgtc
    EXT_texture_filter_anisotropic
    OES_texture_float_linear
    OVR_multiview2
    WEBGL_compressed_texture_s3tc
    WEBGL_compressed_texture_s3tc_srgb
    WEBGL_debug_renderer_info
    WEBGL_debug_shaders
    WEBGL_lose_context

Audio information

System sample rate: 48000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

    WebM Opus (audio/webm; codecs=opus)
    Ogg Opus (audio/ogg; codecs=opus)
    WebM Vorbis (audio/webm; codecs=vorbis)
    Ogg Vorbis (audio/ogg; codecs=vorbis)
    MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
    MP3 (audio/mpeg)
    FLAC (audio/flac)
    PCM WAV (audio/wav; codecs=1)

Supported encode formats:

    WebM Opus (audio/webm; codecs=opus)
    Ogg Opus (audio/ogg; codecs=opus)

Video information

Supported decode formats:

    WebM AV1 (video/webm; codecs=av01.0.00M.08)
    MP4 AV1 (video/mp4; codecs=av01.0.00M.08)
    WebM VP9 (video/webm; codecs=vp9)
    WebM VP8 (video/webm; codecs=vp8)
    Ogg Theora (video/ogg; codecs=theora)
    H.264 (video/mp4; codecs=avc1.42E01E)

Supported encode formats:

    WebM VP8 (video/webm; codecs=vp8)

[colors-oh-my.zip](https://github.com/Scirra/Construct-3-bugs/files/7070115/colors-oh-my.zip)

</details>

**Comment by AshleyScirra** [comment]
It seems to work fine for me: I made a test image saved as PNG in Paint.NET, imported it to Construct, and it preserves the color data exactly. The test image I used is here: 
![test-gradient](https://user-images.githubusercontent.com/768723/131498377-39460d95-ea3b-429f-84ce-e88e22ed7e5a.png)
At the position (200, 100), in both Paint.NET and Construct, the color value is exactly RGB 66, 60, 194.
At position (500, 500), in both Paint.NET and Construct, the color value is exactly RGB 179, 61, 88.
So it seems to be preserving the data exactly correctly.

Construct should import all images to lossless 32-bit PNGs, so the import process should not change anything. I guess other software might use higher bit depths which are not supported, or possibly embed color profile metadata or something like that which gets ignored when importing to Construct, but I'm not sure there's much we could do in either case - ultimately the browser reads and saves the image. Anyway to investigate further we'd need full steps to reproduce, such as the software used, or test images that demonstrate the problem.
